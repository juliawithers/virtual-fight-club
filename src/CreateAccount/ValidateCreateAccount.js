import React from 'react';
import PropTypes from 'prop-types'

export default function ValidateCreateAccount(props) {
  if(props.message) {
      console.log(props.message)
    return (
      <div className="error">{props.message}</div>
    );
  }
  return <></>
}

ValidateCreateAccount.propTypes={
    message: PropTypes.string.isRequired
}