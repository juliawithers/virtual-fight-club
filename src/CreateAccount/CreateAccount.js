import React, { Component } from 'react';
import context from '../context';
import ValidateCreateAccount from './ValidateCreateAccount';

export default class CreateAccount extends Component {
    static contextType = context;
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
        };
    }
   
    handleUserInputErrors = () => {
        const username = this.state.username;
        const password = this.state.password;
        const validateText = /^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{7,15}$/;
        if (!password.match(validateText)) {
            this.setState({
                message: 'Password must be between 7 and 15 characters, and include at least one digit, and one special character'
            });
            return true;
        }
        if (username.length < 4 || username.length > 12) {
            this.setState({
                message: 'Username must be getween 4 and 12 characters'
            });
            return true;
        }
        return false;
    }

    sendUserData = (e) => {
        e.preventDefault();
        const check = this.handleUserInputErrors();
        if (check === false) {
            const object = {
                username: this.state.username,
                passw: this.state.password
            };
            this.context.submitUserInfo(object);
        }

    }

    updateUsername = (e) => {
        const username = e.target.value;
        this.setState({
            username: username
        });
    }

    updatePassword = (e) => {
        const password = e.target.value;
        this.setState({
            password: password
        });
    }

    render() {
        return (
            <section>
                <h2>Create your character today!</h2>
                <form onSubmit={this.sendUserData}>
                    <div>
                        <label htmlFor="username">Username(must be between 4 and 12 characters): </label>
                        <br />
                        <input className="username" type="text" name='username' id='username' onChange={this.updateUsername} />
                    </div>
                    <div>
                        <label htmlFor="password">Password (must be between 7 and 15 characters and include at least one digit and one special character): </label>
                        <br />
                        <input className="password" type="password" name='password' id='password' onChange={this.updatePassword} />
                    </div>
                    <button type='submit'>Sign Up!</button>
                    <ValidateCreateAccount message={this.state.message} />
                </form>
            </section>
        )
    }
}