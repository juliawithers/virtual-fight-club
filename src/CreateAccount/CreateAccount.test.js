import React from 'react';
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReactDOM from 'react-dom';
import CreateAccount from './CreateAccount'

import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

describe(`CreateAccount component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <CreateAccount/>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    })
    it('renders the UI as expected',() =>{
        const wrapper = shallow(<CreateAccount/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})