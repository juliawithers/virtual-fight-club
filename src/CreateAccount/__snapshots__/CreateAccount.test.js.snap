// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount component renders the UI as expected 1`] = `
<section>
  <h2>
    Create your character today!
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="username"
      >
        Username(must be between 4 and 12 characters): 
      </label>
      <br />
      <input
        id="username"
        name="username"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="password"
      >
        Password (must be between 7 and 15 characters and include at least one digit and one special character): 
      </label>
      <br />
      <input
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
      />
    </div>
    <button
      type="submit"
    >
      Sign Up!
    </button>
    <ValidateCreateAccount />
  </form>
</section>
`;
